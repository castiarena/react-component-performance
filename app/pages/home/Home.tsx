import React from 'react';
import Layout from '../../components/Layout';
import Button from '../../components/Button';

const Home = () => (
    <Layout columns={3} gap={40}>
        <div>
            <h1>Hello world</h1>
            <Button>Button</Button>
        </div>
        <div>
            <h1>Hello world</h1>
            <Button>Button</Button>
        </div>
        <div>
            <h1>Hello world</h1>
            <Button>Button</Button>
        </div>
        <div>
            <h1>Hello world</h1>
            <Button>Button</Button>
        </div>
    </Layout>
);

export default Home;
