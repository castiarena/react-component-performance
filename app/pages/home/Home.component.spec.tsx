import React from 'react';
import 'jest-styled-components';
import Home from './Home';
import renderWithTheme from '../../testing-tools/renderWithTheme';

describe('Home component test suite', () => {
    it('smoke test, snapshot only ', () => {
        const { asFragment } = renderWithTheme(<Home />);
        expect(asFragment()).toMatchSnapshot();
    });
});
