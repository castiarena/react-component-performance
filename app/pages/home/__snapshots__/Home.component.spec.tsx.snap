// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test suite smoke test, snapshot only  1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  grid-column-gap: 40px;
  padding: 40px;
}

.c1 {
  border: none;
  padding: 16px 30px;
  border-radius: 24px;
  background-color: #726a95;
  color: #fff;
}

.c1:hover {
  cursor: pointer;
  background-color: #8d83b8;
}

.c1:active,
.c1:focus {
  outline: none;
}

<div
    class="c0"
  >
    <div>
      <h1>
        Hello world
      </h1>
      <button
        class="c1"
      >
        Button
      </button>
    </div>
    <div>
      <h1>
        Hello world
      </h1>
      <button
        class="c1"
      >
        Button
      </button>
    </div>
    <div>
      <h1>
        Hello world
      </h1>
      <button
        class="c1"
      >
        Button
      </button>
    </div>
    <div>
      <h1>
        Hello world
      </h1>
      <button
        class="c1"
      >
        Button
      </button>
    </div>
  </div>
</DocumentFragment>
`;
